### ISSUE TITLE
peer unit bucket is not visible during startup sequence

### ISSUE THREAD

**Issue by sed-i** [canonical/operator#542]
## Environment
multipass (4 cpus 8g ram)
```
$ juju --version
2.9.3-ubuntu-amd64
$ microk8s.kubectl version
Client Version: version.Info{Major:"1", Minor:"20+", GitVersion:"v1.20.7-34+df7df22a741dbc", GitCommit:"df7df22a741dbc18dc3de3000b2393a1e3c32d36", GitTreeState:"clean", BuildDate:"2021-05-12T21:08:20Z", GoVersion:"go1.15.10", Compiler:"gc", Platform:"linux/amd64"}
```

## Steps to reproduce
1. Create a blank charm with `charmcraft init` and add the code below.
2. Deploy a unit and wait for it to become active/idle.
3. Add a second unit.

```python
    def _on_httpbin_pebble_ready(self, event):
        # ...
        relation = self.model.get_relation("replicas")
        relation.data[self.unit]['time_at_pebble_ready'] = str(time.time())

# then use this anywhere during the startup sequence:

    def print_bucket_data(self, prefix: str):
        relation = self.model.get_relation("replicas")
        logger.info("%s: time at pebble_ready: %s", prefix, {unit: relation.data[unit].get('time_at_pebble_ready') for unit in relation.data if isinstance(unit, ops.model.Unit)})

```

## Expected Result
Peers' data buckets are visible to the newly created unit, even during startup sequence.

## Actual Result
Peers' data buckets are not visible to the newly created unit during the startup sequence; only after they become visible.

## Proof
unit-0 has data in its bucket:
```
unit-blank-0: 18:57:21.021 DEBUG unit.blank/0.juju-log Emitting Juju event update_status.
unit-blank-0: 18:57:21.037 INFO unit.blank/0.juju-log on_update_status: time at pebble_ready: {<ops.model.Unit blank/0>: '1622069753.829639'}
```
Then adding unit-1, and during its startup sequence it doesn't see unit-0. The only unit there is unit-1:
```
unit-blank-1: 18:57:32.315 DEBUG unit.blank/1.juju-log Operator Framework 1.2.0 up and running.
unit-blank-1: 18:57:32.319 DEBUG unit.blank/1.juju-log Legacy hooks/config-changed does not exist.
unit-blank-1: 18:57:32.324 DEBUG unit.blank/1.juju-log Using local storage: /var/lib/juju/agents/unit-blank-1/charm/.unit-state.db already exists
unit-blank-1: 18:57:32.329 DEBUG unit.blank/1.juju-log Emitting Juju event config_changed.
unit-blank-1: 18:57:32.336 DEBUG unit.blank/1.juju-log found a new thing: '🎁'
unit-blank-1: 18:57:32.352 INFO unit.blank/1.juju-log on_config_change: time at pebble_ready: {<ops.model.Unit blank/1>: None}
```
After the startup sequence, everything is as expected for both unit-0 and unit-1:
```
unit-blank-1: 18:58:33.952 DEBUG unit.blank/1.juju-log Emitting Juju event update_status.
unit-blank-1: 18:58:33.974 INFO unit.blank/1.juju-log on_update_status: time at pebble_ready: {<ops.model.Unit blank/1>: '1622069856.1295877', <ops.model.Unit blank/0>: '1622069753.829639'}
unit-blank-0: 18:58:34.706 DEBUG unit.blank/0.juju-log Emitting Juju event update_status.
unit-blank-0: 18:58:34.727 INFO unit.blank/0.juju-log on_update_status: time at pebble_ready: {<ops.model.Unit blank/0>: '1622069753.829639', <ops.model.Unit blank/1>: '1622069856.1295877'}

```

**Comment by sed-i** [comment]
Per input from @niemeyer, I confirm that peer unit bucket is visible on "replicas_joined".
@jnsgruk, would be nice to have this added to the [docs](https://juju.is/docs/sdk/relations). 
Closing.
